ironfish/src/network/utils/serializers.ts

```diff
@@ -51,23 +52,23 @@ export function readBlockHeader(reader: bufio.BufferReader): SerializedBlockHead
   const minersFee = (-BigIntUtils.fromBytesLE(reader.readBytes(8))).toString()
   const graffiti = GraffitiSerdeInstance.serialize(reader.readBytes(32))
 
-  return {
+  return new BlockHeader(
     sequence,
-    previousBlockHash,
-    noteCommitment: {
+    Buffer.from(BlockHashSerdeInstance.deserialize(previousBlockHash)),
+    {
       commitment: noteCommitment,
       size: noteCommitmentSize,
     },
-    nullifierCommitment: {
-      commitment: nullifierCommitment,
+    {
+      commitment: NullifierSerdeInstance.deserialize(nullifierCommitment),
       size: nullifierCommitmentSize,
     },
-    target,
-    randomness,
-    timestamp,
-    minersFee,
-    graffiti,
-  }
+    new Target(target),
+    BigInt(randomness),
```

Comment:
You can ditch the `toString` call above, then you can avoid calling `BigInt()` on the string here.

Reply:
Changed